beacon_node/network/src/sync/block_lookups/tests.rs

```diff
@@ -56,91 +87,456 @@ impl TestRig {
+                    NetworkMessage::SendRequest {
+                        peer_id: _,
+                        request: Request::BlobsByRoot(_request),
+                        request_id: RequestId::Sync(SyncRequestId::SingleBlob { id }),
+                    } => Some(*id),
+                    _ => None,
+                })
+                .expect("Expected block request"),
         }
     }
 
+    #[track_caller]
+    fn expect_block_lookup_request(&mut self, for_block: Hash256) -> SingleLookupReqId {
+        self.pop_received_network_event(|ev| match ev {
+            NetworkMessage::SendRequest {
+                peer_id: _,
+                request: Request::BlocksByRoot(request),
+                request_id: RequestId::Sync(SyncRequestId::SingleBlock { id }),
+            } if request.block_roots().to_vec().contains(&for_block) => Some(*id),
+            _ => None,
+        })
+        .expect(&format!("Expected block request for {for_block:?}"))
+    }
+
+    #[track_caller]
+    fn expect_blob_lookup_request(&mut self, for_block: Hash256) -> SingleLookupReqId {
+        self.pop_received_network_event(|ev| match ev {
+            NetworkMessage::SendRequest {
+                peer_id: _,
+                request: Request::BlobsByRoot(request),
+                request_id: RequestId::Sync(SyncRequestId::SingleBlob { id }),
+            } if request
+                .blob_ids
+                .to_vec()
+                .iter()
+                .any(|r| r.block_root == for_block) =>
+            {
+                Some(*id)
+            }
+            _ => None,
+        })
+        .expect(&format!("Expected blob request for {for_block:?}"))
+    }
+
+    #[track_caller]
+    fn expect_block_parent_request(&mut self, for_block: Hash256) -> SingleLookupReqId {
+        self.pop_received_network_event(|ev| match ev {
+            NetworkMessage::SendRequest {
+                peer_id: _,
+                request: Request::BlocksByRoot(request),
+                request_id: RequestId::Sync(SyncRequestId::ParentLookup { id }),
+            } if request.block_roots().to_vec().contains(&for_block) => Some(*id),
+            _ => None,
+        })
+        .expect(&format!("Expected block parent request for {for_block:?}"))
+    }
+
+    #[track_caller]
+    fn expect_blob_parent_request(&mut self, for_block: Hash256) -> SingleLookupReqId {
+        self.pop_received_network_event(|ev| match ev {
+            NetworkMessage::SendRequest {
+                peer_id: _,
+                request: Request::BlobsByRoot(request),
+                request_id: RequestId::Sync(SyncRequestId::ParentLookupBlob { id }),
+            } if request
+                .blob_ids
+                .to_vec()
+                .iter()
+                .all(|r| r.block_root == for_block) =>
+            {
+                Some(*id)
+            }
+            _ => None,
+        })
+        .expect(&format!("Expected blob parent request for {for_block:?}"))
```

Comment:
There's a few instances of this failing the linter
```suggestion
        .unwrap_or_else(|_| panic!("Expected blob parent request for {for_block:?}"))
```